cmd/crypto/kms.go

```diff
@@ -99,6 +99,9 @@ type KMS interface {
 	// keys this method may behave like a NOP and just return the sealedKey
 	// itself.
 	UpdateKey(keyID string, sealedKey []byte, context Context) (rotatedKey []byte, err error)
+
+	// Returns the endpoint, name, authType and error
+	KMSInfo() (endpoint, name, authtype string, err error)
```

Comment:
It is convenient for the consumer of this interface as well as for providing additional information in the future, to return a user-defined data type like `KMSInfo`.
```go
type KMSInfo struct {
  Endpoint string
  Name string
  AuthType string
}
```
and define the above interface method as,
```go
Info() (kmsInfo KMSInfo, err error)
```

Reply:
done. Removed error as it was of no use .